### DISCUSSION TITLE
Differentiate between right alt and left alt in key remap

### DISCUSSION THREAD

**Discussion by vincevargadev** [helix-editor/helix#6479] [Q&A] [answered]
I'd like to add a new keymap and I'd like to differentiate between left alt and right alt (or more precisely, left command and right command buttons as I am on Mac).

What I want to do is:
* hit left alt and `/` to comment out the current line and jump to the end of the next line
* hit right alt and `/` to comment out the current line and jump to the end of the previous line

I didn't find any way to differentiate between the left and right control, alt, etc... keys.

If you could provide any pointers, that would be great, thank you.

**Comment by the-mikedavis** [comment, marked as answer]
This is not currently possible. In the key events we receive from the terminal emulator we only receive the modifier and not which (left/right) modifier is held.

**Reply by eugenesvk** [reply]
What about with the new protocol? Per spec it allows to use different codes for left and right modifiers, is that correct?

**Reply by the-mikedavis** [reply]
Not exactly - modifiers are never left or right, just the key: https://sw.kovidgoyal.net/kitty/keyboard-protocol/#modifiers. There are key press/release events for the modifier keys though which are left or right (https://sw.kovidgoyal.net/kitty/keyboard-protocol/#functional-key-definitions). We could receive these if we set the `KeyboardEnhancementFlags::REPORT_EVENT_TYPES` flag but in order to know whether the modifier is left or right for an event, we would need to roll our own internal modifier system where we track which keys are currently held/pressed.
